server/src/main/java/org/apache/druid/server/coordinator/duty/CompactSegments.java

```diff
@@ -87,6 +91,10 @@ public class CompactSegments implements CoordinatorCustomDuty
   private static final Predicate<TaskStatusPlus> IS_COMPACTION_TASK =
       status -> null != status && COMPACTION_TASK_TYPE.equals(status.getType());
 
+  // An artificial limit imposed just to avoid taking up too many compaction task slots for a single MSQ compaction
+  // task. Can be updated if needed.
+  private static final int MAX_TASK_SLOTS_FOR_MSQ_COMPACTION_TASK = 8;
```

Comment:
Move this constant to `ClientMsqContext`.
Update the default value to 7 maybe (6 + 1). It is currently (7 + 1).